backend/fms_core/services/sample_lineage.py

```diff
@@ -7,12 +7,21 @@ def create_sample_lineage(parent_sample, child_sample, process_measurement):
     errors = []
     warnings = []
 
-    try:
-        sample_lineage = SampleLineage.objects.create(child=child_sample,
-                                                      parent=parent_sample,
-                                                      process_measurement=process_measurement
-                                                      )
-    except ValidationError as e:
-        errors.append(str(e))
+     # Validate parameters
+    if not parent_sample:
+        errors.append(f"Parent sample is required for sample lineage creation.")
+    if not child_sample:
+        errors.append(f"Child sample is required for sample lineage creation.")
+    if not process_measurement:
+        errors.append(f"Process measurement is required for sample lineage creation.")
```

Comment:
Same comment as the one I made previously, I wonder if this is helpful for the user having it displayed in the error messages

Reply:
Services do not pick and choose error messages. If there is an error we identify it and return it. If there is a choice to be made about what is to be reported to the user it is done elsewhere ... The caller can decide to combine all of these with another simplified message (sample lineage will not return an error unless there is something else that is going wrong)... We could just set a generic message at the importer level if there is any sample lineage error ... It is likely accompanied by more useful error messages from parent sample, child sample or sample_measurement.